app/javascript/responseTemplates/responseTemplates.js

```diff
@@ -0,0 +1,225 @@
+      Accept: 'application/json',
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+  })
+    .then(response => response.json())
+    .then(response => {
+      const modResponseHTML =
+        moderatorForTags.length === 0 ? '' : buildModResponseHTML(response);
+      let personalResponseHTML = response
+        .filter(obj => {
+          return obj.type_of === 'personal_comment';
+        })
+        .map(obj => {
+          return `
+            <div class="mod-response-wrapper">
+              <span>${obj.title}</span>
+              <p>${obj.content}</p>
+              <button class="mod-template-button" type="button" data-content="${obj.content}">INSERT</button>
+            </div>
+          `;
+        })
+        .join('');
+      if (personalResponseHTML.length === 0) {
+        personalResponseHTML = `<div class="mod-response-wrapper mod-response-wrapper-empty">
+                                  <p>🤔... It looks like you don't have any templates yet.</p>
+                                  <p>Create templates to quickly answer FAQs or store snippets for re-use.</p>
+                                </div>`;
+      }
+      dataContainer.innerHTML = `
+        ${modResponseHTML}
+        <header><h3>Personal Response Templates</h3></header>
+        ${personalResponseHTML}
+        <a target="_blank" rel="noopener nofollow" href="/settings/response-templates" class="mod-response-create-new">Create new template</a>
+      `;
+
+      if (responsesWrapper) {
+        responsesWrapper.innerHTML = dataContainer.innerHTML
+        addClickListeners(responsesWrapper);
+      }
+    });
+}
+
+export function addReplyObservers() {
+  const targetNodes = Array.from(
+    document.querySelectorAll('div.comment-submit-actions.actions'),
+  );
+  const config = { attributes: false, childList: true, characterData: false };
+  const containerWithData = document.getElementById('response-templates-data');
+
+  const callback = mutationsList => {
+    const { target } = mutationsList[0];
+    if (mutationsList[0].addedNodes.length === 1) {
+      const button = target.querySelector('button.response-templates-button');
+      const responsesWrapper = target.querySelector('.mod-responses-container');
+      if (containerWithData.innerHTML === '') {
+        fetchResponseTemplates(containerWithData, responsesWrapper);
+        responsesWrapper.innerHTML = containerWithData.innerHTML;
+      }
+
+      if (containerWithData.innerHTML !== '') {
+        responsesWrapper.innerHTML = containerWithData.innerHTML;
+        addClickListeners(responsesWrapper)
+      }
+
+      button.addEventListener('click', () => {
+        if (responsesWrapper.style.display === 'none') {
+          responsesWrapper.style.display = 'flex';
+        } else {
+          responsesWrapper.style.display = 'none';
+        }
+      });
+    }
+  };
+
+  const observer = new MutationObserver(callback);
+
+  targetNodes.forEach(node => {
```

Comment:
With this, n target nodes get observed which could be quite expensive. It may make more sense to observe the parent node of all the target nodes.